app/views/cases/show.html.slim

```diff
@@ -47,7 +47,7 @@ div id="case-#{@case.id}"
     = render partial: 'cases/case_status', locals: {case_details: @case}
 
   section.case-info
-    = render partial: 'cases/linked_cases', locals: { case_details: @case }
+    = render partial: 'cases/linked_cases', locals: { case_details: @case } if current_user.can_link_case?(@case)
```

Comment:
I think you should be using the existing Pundit policy `new_case_link?` here to decide whether to show this button or not, and have the logic in there and not in `User.`